### PR TITLE
fru: cleanup redux

### PR DESCRIPTION
**Pull request by pstrinkle**
Fru cleanup.  Clean branch, cherry-picked the changes where possible, recreated where not.  Left out the static cleanup as well.  That can be done more universally later.

### CODE REVIEW

File: lib/ipmi_fru.c [hunk 1]
```diff
@@ -832,11 +845,12 @@ read_fru_area_section(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
...
 		if (rsp->ccode) {
-			/* if we get C7 or C8  or CA return code then we requested too
-			* many bytes at once so try again with smaller size */
-			if ((rsp->ccode == 0xc7 || rsp->ccode == 0xc8 || rsp->ccode == 0xca) &&
-				(--fru_data_rqst_size > 8)) {
-				lprintf(LOG_INFO, "Retrying FRU read with request size %d",
+			/* if we get C7 or C8  or CA return code then we
+			 * requested too many bytes at once so try again with
+			 * smaller size */
+			if (fru_cc_rq2big(rsp->ccode) && (--fru_data_rqst_size > 8)) {
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: Why do we still have `8` here?
Isn't this the same case as in other functions that check for >32 and reduce by 8 ?

File: lib/ipmi_fru.c [hunk 2]
```diff
@@ -4991,24 +4938,22 @@ ipmi_fru_set_field_string_rebuild(struct ipmi_intf * intf, uint8_t fruId,
...
 	for (i = 0;i <= f_index; i++) {
 		fru_field_offset_tmp = fru_field_offset;
-		if (fru_area) {
-			free(fru_area);
-			fru_area = NULL;
-		}
-		fru_area = (uint8_t *) get_fru_area_str(fru_data_old, &fru_field_offset);
+		free_n(&fru_area);
+		fru_area = (uint8_t *)get_fru_area_str(fru_data_old,
+						       &fru_field_offset);
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: Here you have a bad mixture of tabs and spaces. Please indent with tabs to the same level as in previous line, then align with spaces. Currently, there are 2 tabs above and 6 tabs plus 7 spaces here.

File: lib/ipmi_fru.c [hunk 3]
```diff
@@ -726,13 +737,15 @@ read_fru_area(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
...
-			if ((rsp->ccode == 0xc7 || rsp->ccode == 0xc8 || rsp->ccode == 0xca)
-					&& fru->max_read_size > 8) {
-				if (fru->max_read_size > 32) {
-					/* subtract read length more aggressively */
-					fru->max_read_size -= 8;
+			/* if we get C7h or C8h or CAh return code then we
+			 * requested too many bytes at once so try again with
+			 * smaller size */
+			if (fru_cc_rq2big(rsp->ccode)
+			    && fru->max_read_size > FRU_BLOCK_SZ) {
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: For multi-line conditions please put the opening curly brace on a separate line.

File: lib/ipmi_fru.c [hunk 4]
```diff
@@ -726,13 +737,15 @@ read_fru_area(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
...
+				if (fru->max_read_size > FRU_AREA_MAXIMUM_BLOCK_SZ) {
+					/* subtract read length more
+					 * aggressively */
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: Well, this is strange. You break the comment because it's line was 83 characters long with tab size 8, but you leave alone the `if()` line that is 84 characters long. I wouldn't touch the comment. No need to do it.

File: lib/ipmi_fru.c [hunk 5]
```diff
@@ -832,11 +845,12 @@ read_fru_area_section(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
...
 		}
 		if (rsp->ccode) {
-			/* if we get C7 or C8  or CA return code then we requested too
-			* many bytes at once so try again with smaller size */
-			if ((rsp->ccode == 0xc7 || rsp->ccode == 0xc8 || rsp->ccode == 0xca) &&
-				(--fru_data_rqst_size > 8)) {
-				lprintf(LOG_INFO, "Retrying FRU read with request size %d",
+			/* if we get C7 or C8  or CA return code then we
+			 * requested too many bytes at once so try again with
+			 * smaller size */
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: Almost the same here. The next line is just as wide as was the original comment.

File: lib/ipmi_fru.c [hunk 6]
```diff
@@ -4806,14 +4761,14 @@ f_type, uint8_t f_index, char *f_string)
...
+		rc = -1;
 		goto ipmi_fru_set_field_string_out;
 	}
 	memset(fru_data, 0, fru.size);
-	if( read_fru_area(intf ,&fru, fruId, header_offset ,
-					fru_section_len , fru_data) < 0 )
-	{
-		rc = (-1);
+	if (read_fru_area(intf, &fru, fruId, header_offset, fru_section_len,
+			  fru_data)
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: Again, a strange mix of tabs and spaces. Too many tabs.

File: lib/ipmi_fru.c [hunk 7]
```diff
@@ -4806,14 +4761,14 @@ f_type, uint8_t f_index, char *f_string)
...
+	    < 0) {
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: This looks weird. Better move the comparison before the function call (`0 > read_fru_area(...`)

File: lib/ipmi_fru.c [hunk 8]
```diff
@@ -4855,30 +4809,23 @@ f_type, uint8_t f_index, char *f_string)
...
 	}
 	else {
 		printf("String size are not equal, resizing fru to fit new string\n");
-		if(
-				ipmi_fru_set_field_string_rebuild(intf,fruId,fru,header,f_type,f_index,f_string)
-		)
-		{
-			rc = (-1);
+		if (ipmi_fru_set_field_string_rebuild(intf, fruId, fru, header,
+						      f_type, f_index,
```
Comment (AlexanderAmelkin) [inline lib/ipmi_fru.c]: Bad indentation again.

File: lib/ipmi_fru.c (hunk 6 above)
Comment (pstrinkle) [reply lib/ipmi_fru.c]: This change is unrelated to the commit and will be dropped.

File: lib/ipmi_fru.c (hunk 8 above)
Comment (pstrinkle) [reply lib/ipmi_fru.c]: This change is also not related to the commit and will be dropped.

**Comment by pstrinkle** [comment]
PR cleaned up -- dropped extra changes from commits.

File: lib/ipmi_fru.c (hunk 1 above)
Comment (pstrinkle) [reply lib/ipmi_fru.c]: I have been jetlagged since I got back from Christmas, if you mean, let's just swap the number 8 here for the macro because it's the same value it's using, then yeah, let's do that -- just missed this one.  If you think the logic is wrong here, then that's a different story.

File: lib/ipmi_fru.c [hunk 9]
```diff
@@ -3268,8 +3239,8 @@ ipmi_fru_read_to_bin(struct ipmi_intf * intf,
 		return;
 
 	if (rsp->ccode) {
-		if (rsp->ccode == 0xc3)
-			printf ("  Timeout accessing FRU info. (Device not present?)\n");
+		if (rsp->ccode == IPMI_CC_TIMEOUT)
+			printf("  Timeout accessing FRU info. (Device not present?)\n");
```
Comment (pstrinkle) [inline lib/ipmi_fru.c]: todo: undo this change since it's not related to the commit!

File: lib/ipmi_fru.c [hunk 10]
```diff
@@ -3760,76 +3727,66 @@ ipmi_fru_get_multirec(struct ipmi_intf * intf, uint8_t id ,
...
 	return 0;
 }
 
-static int
-ipmi_fru_upg_ekeying(struct ipmi_intf * intf,
-			char * pFileName,
-			uint8_t fruId)
+#define ERR_EXIT do { rc = -1; goto exit; } while(0)
+
+static int ipmi_fru_upg_ekeying(struct ipmi_intf *intf, char *pFileName,
```
Comment (pstrinkle) [inline lib/ipmi_fru.c]: fit this to be:
static
int
ipmi_fru_upg_ekeying

File: lib/ipmi_fru.c [hunk 11]
```diff
@@ -3957,29 +3914,29 @@ ipmi_fru_get_multirec_from_file(char * pFileName, uint8_t * pBufArea,
...
+		return -1;
 	}
 	
 	errno = 0;
 	pFile = fopen(pFileName, "rb");
 	if (!pFile) {
-		lprintf(LOG_ERR, "Error opening file '%s': %i -> %s.", pFileName, errno,
-				strerror(errno));
-		return (-1);
+		lprintf(LOG_ERR, "Error opening file '%s': %i -> %s.",
```
Comment (pstrinkle) [inline lib/ipmi_fru.c]: todo: undo this change.

File: lib/ipmi_fru.c [hunk 12]
```diff
@@ -4712,13 +4666,13 @@ f_type, uint8_t f_index, char *f_string)
...
 		printf(" Device not present (No Response)\n");
-		rc = (-1);
+		rc = -1;
 		goto ipmi_fru_set_field_string_out;
 	}
 	if (rsp->ccode) {
 		printf(" Device not present (%s)\n",
-			val2str(rsp->ccode, completion_code_vals));
-		rc = (-1);
+		       val2str(rsp->ccode, completion_code_vals));
```
Comment (pstrinkle) [inline lib/ipmi_fru.c]: todo: undo this change.

File: lib/ipmi_fru.c [hunk 13]
```diff
@@ -4749,14 +4703,14 @@ f_type, uint8_t f_index, char *f_string)
...
-		rc = (-1);
+		rc = -1;
 		goto ipmi_fru_set_field_string_out;
 	}
 	if (rsp->ccode)
 	{
 		printf(" Device not present (%s)\n",
-				val2str(rsp->ccode, completion_code_vals));
-		rc = (-1);
+		       val2str(rsp->ccode, completion_code_vals));
```
Comment (pstrinkle) [inline lib/ipmi_fru.c]: undo this change.

File: lib/ipmi_fru.c [hunk 14]
```diff
@@ -4806,14 +4759,13 @@ f_type, uint8_t f_index, char *f_string)
...
-		rc = (-1);
+		rc = -1;
 		goto ipmi_fru_set_field_string_out;
 	}
 	memset(fru_data, 0, fru.size);
 	if( read_fru_area(intf ,&fru, fruId, header_offset ,
 					fru_section_len , fru_data) < 0 )
-	{
-		rc = (-1);
+		rc = -1;
```
Comment (pstrinkle) [inline lib/ipmi_fru.c]: bug: add back missing curly brace..

**Comment by pstrinkle** [comment]
Alexander: I went through the PR and noticed a few more issues I want to address, including a bad cherry-pick merge.  I'll reply to this PR when it's worth your time to look at again.

**Comment by pstrinkle** [comment]
Rebased the branch onto 0ca9c66b84a9e20a941d44314070601d7fe16aa6

File: lib/ipmi_fru.c [hunk 15]
```diff
@@ -596,15 +608,14 @@ write_fru_area(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
...
-			if (fru->max_write_size > 8) {
-				fru->max_write_size -= 8;
-				lprintf(LOG_INFO, "Retrying FRU write with request size %d",
-						fru->max_write_size);
-				continue;
-			}
-		} else if(rsp->ccode == 0x80) {
-			rsp->ccode = 0;
+		if (fru_cc_rq2big(rsp->ccode) && fru->max_write_size > FRU_AREA_MAXIMUM_BLOCK_SZ) {
+			fru->max_write_size -= FRU_BLOCK_SZ;
```
Comment (pstrinkle) [inline lib/ipmi_fru.c]: I merged these two conditions, which is a change unrelated to any commit message.

**Comment by pstrinkle** [comment]
Alexander, I removed the last unrelated change (the merging of two conditions).  PTAL at your convenience.

**Comment by AlexanderAmelkin** [comment]
Hurray! This was a long journey, but we did it. Thank you, Patrick for your efforts! I'm merging it now.
CodeFactor is also happy with these changes and says that complexity of the code has reduced.
Great job!
